cmd/api.go

```diff
@@ -45,6 +48,78 @@ func init() {
 
 }
 
+var rich = &cobra.Command{
+	Use:              "richlist [ASSET]",
+	Short:            "Get a list of richest addresses",
+	Example:          "richlist PEG --count=1",
+	PersistentPreRun: always,
+	PreRun:           SoftReadConfig,
+	Args:             cobra.MaximumNArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		cl := srv.NewClient()
+		cl.PegnetdServer = viper.GetString(config.Pegnetd)
+		count, _ := cmd.Flags().GetInt("count")
+		if count == 0 {
+			count = 100
+		}
+
+		if len(args) > 0 {
+			ticker := fat2.StringToTicker(args[0])
+			if ticker == fat2.PTickerInvalid {
+				cmd.PrintErrln(fmt.Errorf("invalid asset specified"))
+				os.Exit(1)
+			}
+
+			assetRich(cl, ticker.String(), count)
+		} else {
+			globalRich(cl, count)
+		}
+	},
+}
+
+func assetRich(cl *srv.Client, asset string, count int) {
+	var params srv.ParamsGetRichList
+	params.Asset = asset
+	params.Count = count
+
+	var res []srv.ResultGetRichList
+	err := cl.Request("get-rich-list", params, &res)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	fmt.Printf("Top %d %s Rich List\n", count, asset)
+	tw := tabwriter.NewWriter(os.Stdout, 0, 0, 2, ' ', 0)
```

Comment:
TIL tab writer. Very cool :+1:

Reply:
yeah the standard libraries have some neat stuff in them, i've been meaning to take some time and just go over a list of what's available that i should be using more often